packages/generator/src/parser/swagger-parser.ts

```diff
@@ -12,5 +12,9 @@ export function parseSwaggerFromPath(swaggerPath: PathLike): SwaggerMetadata {
 }
 
 function parseSwaggerFile(swaggerFile: string): SwaggerMetadata {
-  return JSON.parse(swaggerFile);
+  const swaggerMetaData = JSON.parse(swaggerFile);
+  if (swaggerMetaData.openapi) {
+    swaggerMetaData.definitions = swaggerMetaData?.components?.schemas;
```

Comment:
For openapi 3, the schema has been moved from definitions to components.
Could we check the open api version before doing this line?

Reply:
This is what is happening here. I would like to refrain (at least for now) to parse the version string.